src/bridge-deployer/src/canister_ids.rs

```diff
@@ -0,0 +1,204 @@
+//! A module to perform I/O operations on the `canister_ids.json` file.
+
+mod canisters;
+mod path;
+mod principals;
+
+use std::collections::HashMap;
+
+use candid::Principal;
+use serde::{Deserialize, Serialize};
+use tracing::debug;
+
+pub use self::canisters::Canister;
+pub use self::path::CanisterIdsPath;
+use self::principals::CanisterPrincipal;
+
+/// A struct to map canister names to their principal IDs, which are serialized into file `canister_ids.json`.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct CanisterIds {
+    #[serde(flatten)]
+    canisters: HashMap<Canister, CanisterPrincipal>,
```

Comment:
Why do we need a `CanisterPrincipal` struct? Isn't the principal enough?

Reply:
The field containing the principal is different based on the network type.